universal-application-tool-0.0.1/conf/evolutions/default/17.sql

```diff
@@ -0,0 +1,51 @@
+# --- Create the versions table and migrate existing actives and drafts.
+
+# --- !Ups
+create table versions (
+  id bigserial primary key,
+  lifecycle_stage varchar not null,
+  submit_time timestamp
+);
+
+create table versions_questions (
+  questions_id bigint,
+  versions_id bigint,
+  primary key (questions_id, versions_id)
+);
+
+create table versions_programs (
+  programs_id bigint,
+  versions_id bigint,
+  primary key (programs_id, versions_id)
+);
+
+insert into versions (lifecycle_stage, submit_time) values ('active', current_timestamp);
+
+insert into versions_questions (questions_id, versions_id)
+  select id, 1 from questions
```

Comment:
Question (blocking): do we need to migrate applications that have the old 'versions'?

Reply:
Yep, that's what this does - probably!  Hard to test!  But that's what staging is for, haha.  Oh, unless you mean migrate the ability to go back in time ... in that case, I argue "no it is not worth it."

Reply:
Oh I was worrying about the existing applications, but now I see they don't have a version with it, just lifecycle so I guess they won't be directly affected by this change?